wolfhsm/wh_server_dma.h

```diff
@@ -0,0 +1,114 @@
+#ifndef WH_SERVER_DMA_H_
+#define WH_SERVER_DMA_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include "wolfhsm/wh_server.h"
+
+#define WH_DMA_ADDR_ALLOWLIST_COUNT (10)
+
+struct whServerContext_t;
```

Comment:
Isn't this already in wh_server.h?  Would prefer the typedef'ed one without struct and _t for all parameters below as well.

Reply:
AFAICT there needs to be a forward declaration in `wh_server_dma.h` too because of the bidirectional dependency between the two files. Specifically, `wh_server.h` includes `wh_server_dma.h` before defining the server context (which it must, as the server context has a member of type `whServerDmaContext`). So if that forward declaration goes away then you get undefined type errors in `wh_server_dma.h`. I played around with it a bit and the only other way I could get it to compile cleanly was to typedef `whServerContext` before including `wh_server_dma.h` (which is gross). I figured forward declaring it in `wh_server_dma.h` wasn't a big issue because there is already a bidirectional dependency between the two modules - `wh_server_dma.h` isn't an independent API, but rather just extends the server API but in a separate file for clarity. I think the real solution from an encapsulation perspective would just be to have this all in `wh_server.{c,h}`. Let me know if you think there is a better way to do it and I can revise.